hellabot.go

```diff
@@ -310,13 +313,31 @@ func (bot *Bot) Close() error {
 	return nil
 }
 
-// AddTrigger adds a given trigger to the bots handlers
+// AddTrigger adds a given trigger to the bot's handlers
 func (bot *Bot) AddTrigger(t Trigger) {
+	bot.triggersMu.Lock()
 	bot.triggers = append(bot.triggers, t)
+	bot.triggersMu.Unlock()
+}
+
+// DropTrigger removes a trigger from the bot's handlers
+func (bot *Bot) DropTrigger(t Trigger) bool {
+	bot.triggersMu.Lock()
+	for i, tt := range bot.triggers {
+		if t.Name != "" && t.Name == tt.Name {
+			bot.triggers = append(bot.triggers[:i], bot.triggers[i+1:]...)
+			return true
+		}
+	}
+	bot.triggersMu.Unlock()
```

Comment:
You need to use a `defer` to unlock here because of the `return true` above.

Reply:
That would explain a lot. I've been locally testing and getting weird results.